codegen/tests/simple/types.go

```diff
@@ -23,6 +25,39 @@ type Item struct {
 	Name string `json:"-"`
 }
 
+// PostRequest ...
+type PostRequest struct {
+	Bt *bool            `json:"Bt,omitempty"`
+	Dt convert.JSONTime `json:"Dt"`
+	St string           `json:"St"`
+}
+
+func (t *PostRequest) UnmarshalJSON(data []byte) error {
+	inner := struct {
+		Bt *bool
+		Dt *convert.JSONTime
+		St *string
+	}{}
+	err := json.Unmarshal(data, &inner)
+	if err != nil {
+		return err
+	}
+	if inner.Dt == nil {
+		return errors.New("Dt cannot be nil")
+	}
+
+	if inner.St == nil {
+		return errors.New("St cannot be nil")
+	}
+
+	*t = PostRequest{
+		Dt: *inner.Dt,
```

Comment:
These should be ordered in the same manner as above.

Reply:
That shouldn't affect the code behaviour but ok.